File: 3rd course/2 semester/СПП/Lab8/quadratic.py
def quadratic(a, b, c):
    d = b**2 - 4 * a * c
    if d < 0:
        return "Is Complex"
    elif d == 0:
        x = (-b + d**0.5) / (2 * a)
        return f"{x}"
    else:
        x1 = (-b + d**0.5) / (2 * a)
        x2 = (-b - d**0.5) / (2 * a)
        return f"{x1}\n{x2}"

print(quadratic(1, -3, 2))

File: 3rd course/2 semester/СПП/Lab8/quadratic.pl
quadratic(A, B, C):- 
    D is B*B - 4 * A * C,(
    D < 0 -> write('Is Complex'),nl, false;

    D =:= 0 -> 
    R1 is (- B + sqrt(D))/ (2 * A),
    write(R1), nl ;

    D > 0 ->
    R1 is (- B + sqrt(D))/ (2 * A),
    R2 is (- B - sqrt(D))/ (2 * A),

    write(R1), nl,
    write(R2)).
